11-Backbonejs/05-Backbone-Collections.md

```diff
@@ -2,201 +2,262 @@
 
-We will use the following collection to match our Todo-list application.
+First we'll set up a listener for the `'remove'` event. In `TaskListView.initialize()`:
 
 ```javascript
-// app/js/collections/todolist.js
+var TaskListView = Backbone.View.extend({
+  initialize: function(options) {
+    // ...
 
-TodoManager.Collections.TodoList = Backbone.Collection.extend({
-  initialize: function() {
-    // Event listeners go here
+    // Re-render the whole list when the Collection changes
+    this.listenTo(this.model, 'update', this.render);
 
+    // When a Model is removed from the Collection, remove the
+    // corresponding card from our list.
+    this.listenTo(this.model, 'remove', this.removeTask);
   },
-  model: TodoManager.Models.Todo
+	// ...
 });
 ```
 
-First we need to build a template to render the collection into.
-
-```html
-<script type="text/template" id="tpl-todolist">
-      <h2 class="page-header text-center">Things To Do</h2>
-      <p class="text-center">
-        <a href="#todo/new" class="btn btn-lg btn-outline">Add Item</a>
-      </p>
-      <article class="media-list media-list row small-up-1 medium-up-2 large-up-4 todolist-container "></article>
-</script>
-```
-
-And then the view to render the collection using the template.  Notice we use `that.$('.media-list)`.  Backbone provides a jQuery object you can use to select elements within the view's HTML.  We use that to select an article within the HTML to append each items' views to.  
-
+Next, our event handler, `removeTask()`. [Again from the docs](http://backbonejs.org/#Events-catalog), the handler will be supplied with three arguments: `model`, `collection`, and `options`. `model` is the only one we need. Our strategy will be to create a new card list containing every card where the card's model doesn't match the removed model.
 
 ```javascript
-TodoManager.Views.TodoList = Backbone.View.extend( {
-  tagName: 'section',
-  className: 'media no-bullet column',
-  template: _.template($('#tpl-todolist').html()),
-  initialize:  function() {
-    // put event listeners here!
-  },
-  render:  function() {
-    var that = this;
-      // Clear the HTML with the fresh template when rendering
-    this.$el.html(this.template());
-    _.each( that.collection.models, function(item) {
-
-      var myTodoView = new TodoManager.Views.Todo ({
-          model: item
-        });
-        // Select the Media-List class article inside the template
-        // then append the item's view.
-      that.$('.media-list').append(myTodoView.render().el);
+var TaskListView = Backbone.View.extend({
+  // ...
+
+  removeTask: function(task) {
+    var filteredList = [];
+    this.cardList.forEach(function(card) {
+      if (card.model != task) {
+        filteredList.push(card);
+      }
     });
-
-    return this;
+    this.cardList = filteredList;
   }
 });
 ```
 
-And then in `app/js/app.js` we can create the collection and view and render it.
-
+Or more succinctly
```

Comment:
A couple of students have commented to me that they have found it very confusing when we've shown them how to do a thing one way and then immediately show them how to do it a different, but equivalent, way.

I think we should probably stick with the first of these two `removeTask()` functions, because it will probably be more familiar to the majority of students.